sourcerer-esjc/src/main/java/org/elder/sourcerer2/esjc/EventStoreEsjcEventRepository.kt

```diff
@@ -0,0 +1,553 @@
+                } else if (ex.cause is CannotEstablishConnectionException
+                        || ex.cause is ClusterException
+                        || ex.cause is ConnectionClosedException
+                        || ex.cause is OperationTimedOutException
+                        || ex.cause is MaximumSubscribersReachedException
+                        || ex.cause is RetriesLimitReachedException
+                        || ex.cause is ServerErrorException) {
+                    throw RetriableEventWriteException(ex.cause)
+                } else {
+                    logger.warn("Unrecognized event store exception writing events", ex.cause)
+                    throw RetriableEventWriteException(ex.cause)
+                }
+            } else if (ex.cause is RuntimeException) {
+                logger.warn("Unrecognized runtime exception writing events", ex.cause)
+                throw RetriableEventWriteException(ex.cause)
+            } else {
+                logger.warn("Unrecognized exception writing events", ex.cause)
+                throw RetriableEventWriteException(
+                        "Internal error writing events",
+                        ex.cause)
+            }
+        } catch (ex: TimeoutException) {
+            throw RetriableEventWriteException("Timeout writing events", ex.cause)
+        }
+
+    }
+
+    private inner class EmitterListener(
+            private val emitter: FluxEmitter<EventSubscriptionUpdate<T>>,
+            private val name: String) : CatchUpSubscriptionListener {
+
+        override fun onEvent(subscription: CatchUpSubscription, event: ResolvedEvent) {
+            logger.debug("Incoming message in {}: {}", name, event)
+            emitter.next(EventSubscriptionUpdate.ofEvent(fromEsEvent(event)))
+        }
+
+        override fun onClose(
+                subscription: CatchUpSubscription?,
+                reason: SubscriptionDropReason?,
+                exception: Exception?) {
+            if (exception != null) {
+                logger.error(
+                        "Subscription $name failed with reason $reason",
+                        exception)
+            } else {
+                logger.error(
+                        "Subscription {} failed with reason {} and no exception",
+                        name,
+                        reason)
+            }
+
+            emitter.fail(EventStoreSubscriptionStoppedException(reason, exception))
+        }
+
+        override fun onLiveProcessingStarted(subscription: CatchUpSubscription?) {
+            logger.info("Live processing started for {}!", name)
+            emitter.next(EventSubscriptionUpdate.caughtUp<T>())
+        }
+    }
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(EventStoreEsjcEventRepository::class.java)
+
+        private const val MAX_MAX_EVENTS_PER_READ = 4095
+        private const val DEFAULT_TIMEOUT_MILLIS = 30 * 1000.toLong()
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)